src/zarr/storage/zip.py

```diff
@@ -131,13 +132,19 @@ def with_mode(self, mode: ZipStoreAccessModeLiteral) -> Self:  # type: ignore[ov
         raise NotImplementedError("ZipStore cannot be reopened with a new mode.")
 
     def __str__(self) -> str:
-        return f"zip://{self.path}"
+        # lets try https://github.com/zarr-developers/zeps/pull/48/files
+        return f"file:///{self.file_path}|zip://{self.path}"
```

Comment:
I think you have too many slashes after `file:` because `file_path` also starts with a slash, right?

Also you should percent-encode both `file_path` and `path` for consistency with my ZEP proposal and tensorstore, and to avoid ambiguity since `|` is a legal path character except on Windows.

Reply:
thanks for catching this!